### PR TITLE
Some unit tests for Parser

### PR DESCRIPTION
**Pull request by growse**
Also pulled some preferences out of static to make them more testable.

### CODE REVIEW

File: project/app/src/test/java/org/owntracks/android/support/ParserTest.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        mockStatic(App.class);
+        org.powermock.api.mockito.PowerMockito.when(App.getPreferences()).thenReturn(testPreferences);
+        when(testPreferences.getEncryptionKey()).thenReturn("testEncryptionKey");
+        when(encryptionProvider.isPayloadEncryptionEnabled()).thenReturn(true);
+    }
+
+    @Test
+    public void ParserCorrectlyConvertsLocationToPrettyJSON() throws Exception {
+        Parser parser = new Parser(null);
+        String expected = "{\n" +
```
Comment (CloCkWeRX) [inline project/app/src/test/java/org/owntracks/android/support/ParserTest.java]: Thats a bit hard to read with all of the escaping, maybe better with triple quotes or stored on disk/loaded into a JSON parser?

Comment (growse) [reply project/app/src/test/java/org/owntracks/android/support/ParserTest.java]: Fair point - I'll look into making it more readable.

**Comment by growse** [comment]
Turns out Java doesn't have any nice way of doing multiline strings, so I pulled into a resource file.

**Comment by binarybucks** [comment]
Much better. Good to merge?

**Comment by growse** [comment]
I think so. The ultimate aim is to eventually get as much coverage as possible. I'll probably look to add more tests in class by class in the future.

**Pull request merged**